src/index.js

```diff
@@ -0,0 +1,189 @@
+const fileListMetadata = document.getElementById("fileListMetadata");
+const fileNum = document.getElementById("fileNum");
+
+uploaderInput.addEventListener("change", validateFileSizeOnSubmit);
+
+["dragenter", "dragover"].forEach((eventName) => {
+  dropArea.addEventListener(eventName, highlight, false);
+});
+
+function highlight(event) {
+  event.preventDefault();
+  event.stopPropagation();
+
+  dropArea.classList.add("highlight");
+}
+
+["dragleave", "drop"].forEach((eventName) => {
+  dropArea.addEventListener(eventName, unhighlight, false);
+});
+
+function unhighlight(event) {
+  event.preventDefault();
+  event.stopPropagation();
+
+  dropArea.classList.remove("highlight");
+}
+
+dropArea.addEventListener("drop", handleDrop, false);
+
+function handleSubmit(event) {
+  event.preventDefault();
+  submitButton.disabled = true;
+
+  // status.textContent = "⏳ Pending...";
+
+  const form = event.currentTarget;
+  const url = new URL(form.action);
+  const formMethod = form.method;
+  const formData = new FormData(form);
+  const progressBar = document.getElementById("progressBar");
+
+  let xhr = new XMLHttpRequest();
+
+  xhr.upload.onprogress = function (event) {
+    status.textContent = `Uploaded ${event.loaded} bytes of ${event.total}`;
+
+    let percent = (event.loaded / event.total) * 100;
+    progressBar.value = Math.round(percent);
+  };
+
+  xhr.onloadend = function () {
+    if (xhr.status === 200) {
+      status.textContent = "✅ Success";
+      progressBar.value = 0;
+      submitButton.disabled = false;
+
+      getFilesMetadata(uploaderInput.files);
+    } else {
+      status.textContent = "❌ Error";
+      progressBar.value = 0;
+      submitButton.disabled = false;
+    }
+  };
+
+  xhr.open(formMethod, url);
+  xhr.send(formData);
+}
+
+function handleDrop(event) {
+  event.preventDefault();
+  event.stopPropagation();
+  submitButton.disabled = true;
+
+  // status.textContent = "⏳ Pending...";
+
+  const fileList = event.dataTransfer.files;
+  const form = event.currentTarget.getElementsByTagName("form")[0];
+  const url = new URL(form.action);
+  const formMethod = form.method;
+  const progressBar = document.getElementById("progressBar");
+
+  if (validateFileSizeOnDrop(fileList)) {
+    let xhr = new XMLHttpRequest();
+
+    xhr.upload.onprogress = function (event) {
+      status.textContent = `Uploaded ${event.loaded} bytes of ${event.total}`;
+
+      let percent = (event.loaded / event.total) * 100;
+      progressBar.value = Math.round(percent);
+    };
+
+    xhr.onloadend = function () {
+      if (xhr.status === 200) {
+        status.textContent = "✅ Success";
+        progressBar.value = 0;
+        submitButton.disabled = false;
+
+        getFilesMetadata(fileList);
+      } else {
+        status.textContent = "❌ Error";
+        progressBar.value = 0;
+        submitButton.disabled = false;
+      }
+    };
+
+    xhr.open(formMethod, url);
+    xhr.send(event.dataTransfer.files);
```

Comment:
17 bytes because `event.dataTransfer.files.toString().length === 17` :)

You need to create a [FormData instance](https://developer.mozilla.org/en-US/docs/Web/API/FormData/FormData), append files (one file - one field) and then pass it to the `xhr.send`